src/Razor/test/Microsoft.VisualStudio.LanguageServerClient.Razor.Test/LSPRazorProjectHostTest.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Razor.Workspaces;
+using Microsoft.VisualStudio.LanguageServer.Client;
+using Microsoft.VisualStudio.Threading;
+using Moq;
+using Xunit;
+
+namespace Microsoft.VisualStudio.LanguageServerClient.Razor
+{
+    public class LSPRazorProjectHostTest
+    {
+        public LSPRazorProjectHostTest()
+        {
+            var joinableTaskContext = new JoinableTaskContextNode(new JoinableTaskContext());
+            JoinableTaskContext = joinableTaskContext.Context;
+        }
+
+        private JoinableTaskContext JoinableTaskContext { get; }
+
+        [Fact]
+        public async Task LoadAsync_NoopsWhenLSPEditorFeatureNotAvailable()
+        {
+            // Arrange
+            var featureDetector = Mock.Of<LSPEditorFeatureDetector>(f => f.IsLSPEditorFeatureEnabled() == false);
+            var broker = new Lazy<ILanguageClientBroker>();
+            var client = new Lazy<ILanguageClient, IDictionary<string, object>>(
+                () => throw new NotImplementedException(), new Dictionary<string, object>());
+            var languageClients = new[] { client };
+
+            var host = new LSPRazorProjectHost(JoinableTaskContext, featureDetector, broker, languageClients);
+
+            // Act & Assert (does not throw)
+            await host.LoadAsync().ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task LoadAsync_LSPEditorFeatureAvailable_LoadsClients()
```

Comment:
What happens when one of the clients throws on Load?

Reply:
I believe that'd be a catastrophic failure. Is this something we want to safeguard against / catch?

Reply:
We should do what LSP platform does when it loads the clients. I don't remember them doing any special handling for this. It would be good to double check.

Reply:
Confirmed.